services/QuillLMS/app/services/google_integration/user.rb

```diff
@@ -43,10 +43,11 @@ def update_or_initialize
 
   private def auth_credential_attributes(user)
     {
-      provider:      'google',
-      refresh_token: profile.refresh_token || user&.auth_credential&.refresh_token,
-      expires_at:    profile.expires_at,
       access_token:  profile.access_token,
+      expires_at:    profile.expires_at,
+      refresh_token: profile.refresh_token || user&.auth_credential&.refresh_token,
```

Comment:
This is orthogonal to this PR, but can we get rid of a code path and just do 

`refresh_token: profile.refresh_token` ?

Reply:
Also, this line will need to be updated if you implement Dan's suggestion of adding `has_one` associations in `User.rb` for each STI class

Reply:
Good call.  I will remove that.